odoo/cli/obfuscate.py

```
@@ -0,0 +1,246 @@
+            sys.exit(0)
+        return True
+
+    def run(self, cmdargs):
+        parser = odoo.tools.config.parser
+        group = optparse.OptionGroup(parser, "Populate Configuration")
+        group.add_option('--pwd', dest="pwd", default=False, help="Cypher password")
+        group.add_option('--fields', dest="fields", default=False, help="List of table.columns to obfuscate/unobfuscate: table1.column1,table2.column1,table2.column2")
+        group.add_option('--exclude', dest="exclude", default=False, help="List of table.columns to exclude from obfuscate/unobfuscate: table1.column1,table2.column1,table2.column2")
+        group.add_option('--file', dest="file", default=False, help="File containing the list of table.columns to obfuscate/unobfuscate")
+        group.add_option('--unobfuscate', action='store_true', default=False)
+        group.add_option('--allfields', action='store_true', default=False, help="Used in unobfuscate mode, try to unobfuscate all fields. Cannot be used in obfuscate mode. Slower than specifying fields.")
+        group.add_option('--vacuum', action='store_true', default=False, help="Vacuum database after unobfuscating")
+        group.add_option('--pertablecommit', action='store_true', default=False, help="Commit after each table instead of a big transaction")
+
+        parser.add_option_group(group)
+        if not cmdargs:
+            sys.exit(parser.print_help())
+
+        try:
+            opt = odoo.tools.config.parse_config(cmdargs)
+            if opt.allfields and not opt.unobfuscate:
+                _logger.error("--allfields can only be used in unobfuscate mode")
+                sys.exit("ERROR: --allfields can only be used in unobfuscate mode")
+            self.dbname = odoo.tools.config['db_name']
+            self.registry = odoo.registry(self.dbname)
+            with self.registry.cursor() as cr:
+                self.cr = cr
+                self.begin()
+                if self.check_pwd(opt.pwd):
```

This line will fail with the weird error if pwd is not set : 

```
ERROR: ERREUR:  la fonction pgp_sym_decrypt(bytea, boolean) n'existe pas
LIGNE 1 : ...CASE WHEN starts_with("value", 'odoo_cyph_') THEN pgp_sym_de...

```

It could be better to raise an error if this required argument is not provided. See proposal, above.